Yonashenok/ExpensePal
# Ruby on Rails capstone project - Budget app

## Summary 
[@Yonashenok](https://github.com/Yonashenok) I have completed the implementation of [Budget app](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md) on the Budget app according to project [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md). 

The following tasks have been implemented: 
- [x] simple page with the name of your app
- [x] Login page and registration page using devise
- [x] Home page (categories page)
   -  user logs in, they are presented with the categories page
   -  user can see their name, icon and the total amount of all the transactions
   - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category
   - button "add a new category" at the bottom that brings the user to the page to create a new category
- [x] Transactions page
    - The list of transactions is presented, ordered by the most recent  
    - The top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
 - [x] "Add a new category" page
    -  The user fills out a form to create a new category, indicating their name and icon
    - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success
    - When the user clicks on the "Back" button (<), the user navigates to the home page
 - [x]  Add a new transaction" page
    -  he user fills out a form to create a new transaction
    - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
    - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category
- [x] Testing requirements
   -  Create unit and integration tests for all the most important components of your RoR application 




[Technical requirements](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#technical-requirements)
- Should use devise for authentication.
- Should validate all user input to make sure that anyone with bad intentions - - cannot compromise your app.
- Can use a view template engine of your choice (.erb, .slim, .haml).
- Project should be deployed and accessible online.
- Database schema should reflect the following structure ERD:


 


In addition, we have fulfilled the following default requirements:

* Set up the repository on GitHub and use Gitflow.
* Set up Devise for authentication.
* Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
* Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
* Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
* Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
* Set up RSpec and Capybara testing libraries.




we will be  looking forward to your feedback.
Thank you!
